### DISCUSSION TITLE
How to install Android build tools 33.0.2?

### DISCUSSION THREAD

**Discussion by anniewey** [codemagic-ci-cd/codemagic-docs#1528] [Q&A]
Hi, my project is using `33.0.2` but I need to install the build tools in codemagic before building cause I need to make changes to the build tools folder due to this error on macos: `Installed Build Tools revision 33.0.2 is corrupted. Remove and install again using the SDK Manager.` (Ref: build 6426a462ed0b18a3811ceaef)

So I need to rename `d8` to `dx` in the build folder using this [solution](https://support.hcltechsw.com/csm?id=kb_article&sysparm_article=KB0101300)
But when i `ls $ANDROID_SDK_ROOT/build-tools` only have version up to `33.0.0`

My intention is to rename the file in build-tools, before running gradle build
```
//- add a new step to install build-tools 33.0.2 first
 - name: Fix Android SDK build tools
        script: |
          mv $ANDROID_SDK_ROOT/build-tools/33.0.2/d8 $ANDROID_SDK_ROOT/build-tools/33.0.2/dx
          mv $ANDROID_SDK_ROOT/build-tools/33.0.2/lib/d8.jar $ANDROID_SDK_ROOT/build-tools/33.0.2/lib/dx.jar
- name: Build Android App Bundle
        working_directory: android
        script: |
          cd "$CM_BUILD_DIR/android"
          ./gradlew bundleRelease
``

**Comment by icarusdust** [comment]
Hi @anniewey, you can install `33.0.2` by executing the following script at the top of you scripts section:

```
set -e
set -x
JAVA_HOME=/Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home
$ANDROID_SDK_ROOT/tools/bin/sdkmanager --install "build-tools;33.0.2"
```

**Reply by anniewey** [reply]
Thanks! Will try it out.
I had downgraded to `33.0.0` as I need to do a quick release last week.
